actionmailer/test/mailers/lazy_proc_mailer.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class LazyProcMailer < ActionMailer::Base
+  default from: -> { flunk }, to: -> { flunk }
+
+  def welcome
+    mail from: "overridden-from@example.com", to: "overridden-to@example.com"
+  end
+end
```

Comment:
Maybe we could reuse `ProcMailer` instead and use a proc for `to`:
```
class ProcMailer < ActionMailer::Base
  default to: -> { "system@test.lindsaar.net" },
  ...
```
https://github.com/rails/rails/blob/main/actionmailer/test/mailers/proc_mailer.rb#L4

Reply:
I think I need the `LazyProcMailer` here. 

What I want to test here is that the default procs for `from` and `to` are *never* evaluated in the process (and so the flunk never raises an error). This test allows me to make sure about that. 

What do you think? Maybe the name `LazyProcMailer` is confusing and I should change it to something like `NeverEvaluatedDefaultHeadersMailer` to make it more precise?

Reply:
I see the test class also has a with_default method you can use:
```ruby  
  test "proc default values are not evaluated when overridden" do
    with_default BaseMailer, to: -> { flunk } do
        email = BaseMailer.welcome(to: "overridden-from@example.com")
        assert_equal "overridden-from@example.com", email.to
      end
    end
  end
  ```

Reply:
Yes nice! 
I made the change as well, it should be all good now! 

Thank you very much again for your review 🙏